lib/debezium/decimal_test.go

```diff
@@ -56,6 +57,34 @@ func mustEncodeAndDecodeDecimal(value string, scale uint16) string {
 	return out
 }
 
+func mustParseDecimalFromString(in string) *apd.Decimal {
+	out, _, err := apd.NewFromString(in)
+	if err != nil {
+		panic(err)
+	}
+	return out
+}
+
+func TestDecimalWithNewExponent(t *testing.T) {
+	assert.Equal(t, "0", decimalWithNewExponent(apd.New(0, 0), 0).Text('f'))
+	assert.Equal(t, "00", decimalWithNewExponent(apd.New(0, 1), 1).Text('f'))
+	assert.Equal(t, "0", decimalWithNewExponent(apd.New(0, 100), 0).Text('f'))
+	assert.Equal(t, "00", decimalWithNewExponent(apd.New(0, 0), 1).Text('f'))
+	assert.Equal(t, "0.0", decimalWithNewExponent(apd.New(0, 0), -1).Text('f'))
+
+	// Same exponent:
+	assert.Equal(t, "12.349", decimalWithNewExponent(mustParseDecimalFromString("12.349"), -3).Text('f'))
+	// More precise exponent:
+	assert.Equal(t, "12.3490", decimalWithNewExponent(mustParseDecimalFromString("12.349"), -4).Text('f'))
+	assert.Equal(t, "12.34900", decimalWithNewExponent(mustParseDecimalFromString("12.349"), -5).Text('f'))
+	// Lest precise exponent:
+	// Extra digits should be truncated rather than rounded.
```

Comment:
Should we add a test case where the first truncated digit is >= 5 to make sure it will be truncated instead of rounded up?

Reply:
Isn't that the next line:
> assert.Equal(t, "12.34", decimalWithNewExponent(mustParseDecimalFromString("12.349"), -2).Text('f'))

Reply:
Oh yeah you're right, I missed that it was truncating before the 9. Carry on!